File: Python/Eghit.py
# Assume you have a method isSubstring which checks if one word is a substring of another.
# Given two strings, s1 and s2, write code to check if s2 is a rotation of s1 using only one call to isSubstring
# (i.e., "waterbottle" is a rotation of "erbottlewat").

def isRotate(str1, str2):
    con = str1 + str1
    if str2 in con:
        print("TRUE")
    else:
        print("FALE")

str1 = input("Enter first string: ")
str2 = input("Enter second string: ")
isRotate(str1, str2)

File: Interview/src/array_string/Eghit.java
/*
 * Assume you have a method isSubstring which checks if one word is a substring of another.
 * Given two strings, s1 and s2, write code to check if s2 is a rotation of s1 using only one call to isSubstring
 * (i.e., "waterbottle" is a rotation of "erbottlewat").
 */
package array_string;

/**
 *
 * @author Milad
 */
public class Eghit {
    public static void main(String[] args){
        System.out.println(isRotate("milad sabouri", "ad sabouri"));
    }
    
    public static boolean isRotate(String str1, String str2){
        String con = str1 + str1;
        if(con.contains(str2))
            return true;
        else
            return false;
    }
}
